.github/workflows/build_and_test.yml

```diff
@@ -379,6 +379,7 @@ jobs:
       SKIP_PACKAGING: true
       METASPACE_SIZE: 1g
       BRANCH: ${{ inputs.branch }}
+      SPARK_ANSI_SQL_MODE: true
```

Comment:
Check ansi is true , will revert this file after the test passes.

Reply:
![image](https://github.com/apache/spark/assets/1475305/eff3f8a1-d4e2-40f7-ac3f-b04b6bf058da)
Passed